Git-commit: 4f19d19906c3c020376e8f1f97bb862fcd406678
From: Xie XiuQi <xiexiuqi@huawei.com>
Date: Fri, 26 Feb 2021 20:21:37 +0800
Subject: [PATCH] arm64/mpam: use snprintf instead of sprintf
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.5.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

hulk inclusion
category: bugfix
bugzilla: 48265
CVE: NA

--------------------------------

snprintf is more safe than sprintf, it is recommended
to use snprintf.

Signed-off-by: Xie XiuQi <xiexiuqi@huawei.com>
Reviewed-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Yang Yingliang <yangyingliang@huawei.com>
Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm64/kernel/mpam_ctrlmon.c | 2 +-
 fs/resctrlfs.c                   | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/mpam_ctrlmon.c b/arch/arm64/kernel/mpam_ctrlmon.c
index 02623c1c23e8..86e50fd19309 100644
--- a/arch/arm64/kernel/mpam_ctrlmon.c
+++ b/arch/arm64/kernel/mpam_ctrlmon.c
@@ -415,7 +415,7 @@ static int mkdir_mondata_subdir(struct kernfs_node *parent_kn,
 	md.u.partid = prgrp->closid;
 	md.u.pmg = prgrp->mon.rmid;
 
-	sprintf(name, "mon_%s_%02d", r->name, d->id);
+	snprintf(name, sizeof(name), "mon_%s_%02d", r->name, d->id);
 	kn = __kernfs_create_file(parent_kn, name, 0444,
 				  GLOBAL_ROOT_UID, GLOBAL_ROOT_GID, 0,
 				  &kf_mondata_ops, md.priv, NULL, NULL);
diff --git a/fs/resctrlfs.c b/fs/resctrlfs.c
index b7b52db4c0a6..267c3fc92000 100644
--- a/fs/resctrlfs.c
+++ b/fs/resctrlfs.c
@@ -200,7 +200,7 @@ static int resctrl_group_create_info_dir(struct kernfs_node *parent_kn)
 	for_each_resctrl_resource(r) {
 		if (r->mon_enabled) {
 			fflags =  r->fflags | RF_MON_INFO;
-			sprintf(name, "%s_MON", r->name);
+			snprintf(name, sizeof(name), "%s_MON", r->name);
 			ret = resctrl_group_mkdir_info_resdir(r, name, fflags);
 			if (ret)
 				goto out_destroy;
-- 
2.31.1

